reference/sql_workarounds.rst

```diff
@@ -0,0 +1,141 @@
+.. _workarounds:
+
+SQL Workarounds
+===============
+
+Citus supports most, but not all, SQL statements directly. Its SQL support continues to improve. Also many of the unsupported features have workarounds; below are a number of the most useful.
+
+Subqueries
+----------
+
+In analytics it is common to model information using a "star schema." This is a collection of dimensional tables (e.g. observations, statistics) referencing a table of facts (e.g. people, pages or events). Furthermore, a common form of query examines dimensions for a subset of the facts. Examples of these kind of queries:
+
+* "Get aggregated statistics about football players whose passes exceed three hundred yards."
+* "Which other pages do people visit who view a certain page?"
+* "Count the number of distinct sessions for the top twenty-five most visited blog posts."
+* "What is the number of people in companies among the top fifty product areas?"
+
+The general form of the query in SQL is
+
+.. code-block:: sql
+
+  select dim_a, dim_b, dim_c
+    from dimensions
+   where fact_id in (select id from facts where ...conditions...)
+     and ...conditions...;
+
+Citus does not allow subqueries in the WHERE clause so we must choose a workaround.
+
+Workaround 1. Generate explicit WHERE-IN expression
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Pros
+    * Able to distribute the query across shards
+* Cons
+    * Constructed query not suitable when facts subquery returns large number of results
+    * Somewhat hard to read, requires building custom SQL
+
+In this technique we use PL/pgSQL to construct and execute one statement based on the results of another.
+
+.. code-block:: sql
+
+  -- create temporary table with results
+  do language plpgsql $$
+    declare facts_array integer[];
+  begin 
+    execute 'select id from facts where ...conditions...'
+       into facts_array;
+    execute format(
+      'create temp table results_temp as '
+      'select dim_a, dim_b, dim_c'
+      '  from dimension'
+      ' where fact_id = any(array[%s])'
+      '   and ...conditions...',
+      array_to_string(facts_array, ','));
+  end;
+  $$;
+
+  -- read results, remove temp table
+  select * from results_temp;
+  drop table results_temp;
+
+
+Workaround 2. Join on Master
```

Comment:
You don't require a join here. You could directly run a subquery in where for the local temp tables which you create. It would work, because the local tables wouldn't be distributed.